pkg/pipeline/resource_pipeline_project_integration.go

```diff
@@ -0,0 +1,282 @@
+		var formJSONValues []FormJSONValues
+		keyValues := d.Get(key).([]interface{})
+		for _, keyValue := range keyValues {
+			idx := keyValue.(map[string]interface{})
+			formJSONValue := FormJSONValues{
+				Label: idx["label"].(string),
+				Value: idx["value"].(string),
+			}
+			formJSONValues = append(formJSONValues, formJSONValue)
+		}
+		return formJSONValues
+	}
+
+	var packFormJSONValues = func(d *schema.ResourceData, schemaKey string, formJSONValues []FormJSONValues) []error {
+		setValue := mkLens(d)
+		var keyValues []interface{}
+		for _, idx := range formJSONValues {
+			keyValue := map[string]interface{}{
+				"label": idx.Label,
+				"value": idx.Value,
+			}
+			keyValues = append(keyValues, keyValue)
+		}
+		errors := setValue(schemaKey, keyValues)
+		return errors
+	}
+
+	var unpackProjectIntegration = func(data *schema.ResourceData) (ProjectIntegration, error) {
+		d := &ResourceData{data}
+
+		projectIntegration := ProjectIntegration{
+			Name:                  d.getString("name"),
+			ProjectId:             d.getInt("project_id"),
+			MasterIntegrationId:   d.getInt("master_integration_id"),
+			MasterIntegrationName: d.getString("master_integration_name"),
+			Environments:          d.getList("environments"),
+			IsInternal:            d.getBool("is_internal"),
+			Project:               unpackProject(d, "project"),
+			FormJSONValues:        unpackFormJSONValues(d, "form_json_values"),
+		}
+		return projectIntegration, nil
+	}
+
+	var packProjectIntegration = func(d *schema.ResourceData, projectIntegration ProjectIntegration) diag.Diagnostics {
+		var errors []error
+		setValue := mkLens(d)
+
+		errors = setValue("project_id", projectIntegration.ProjectId)
+		errors = append(errors, setValue("name", projectIntegration.Name)...)
+		errors = append(errors, setValue("project_id", projectIntegration.ProjectId)...)
+		errors = append(errors, setValue("master_integration_id", projectIntegration.MasterIntegrationId)...)
+		errors = append(errors, setValue("master_integration_name", projectIntegration.MasterIntegrationName)...)
+		errors = append(errors, setValue("environments", projectIntegration.Environments)...)
+		errors = append(errors, setValue("is_internal", projectIntegration.IsInternal)...)
+		errors = append(errors, packProject(d, "project", projectIntegration.Project)...)
+		errors = append(errors, packFormJSONValues(d, "form_json_values", projectIntegration.FormJSONValues)...)
+
+		if len(errors) > 0 {
+			return diag.Errorf("failed to pack project integration %q", errors)
+		}
+
+		return nil
+	}
+
+	var readProjectIntegration = func(ctx context.Context, data *schema.ResourceData, m interface{}) diag.Diagnostics {
+		log.Printf("[DEBUG] readProjectIntegration")
+		projectIntegration := ProjectIntegration{}
+		resp, err := m.(*resty.Client).R().
+			SetResult(&projectIntegration).
+			Get(projectIntegrationsUrl + "/" + data.Id())
+		log.Println("[DEBUG] projectIntegration body: ", string(json.RawMessage(resp.Body())))
```

Comment:
We might consider putting this code (or something like it) into the rest `BeforeRequest` and `AfterReponse` - Right now the code will only dump the body on error -